cumulus/client/cirrus-executor/src/aux_schema.rs

```diff
@@ -47,18 +59,28 @@ pub(super) fn write_execution_receipt<Backend: AuxStore, Block: BlockT>(
 		new_first_saved_receipt = block_number.saturating_sub((PRUNING_DEPTH - 1).saturated_into());
 
 		let mut keys_to_delete = vec![];
-		let mut to_delete_start = first_saved_receipt;
-		while to_delete_start < new_first_saved_receipt {
-			keys_to_delete.push((EXECUTION_RECEIPT_KEY, to_delete_start).encode());
-			to_delete_start = to_delete_start.saturating_add(One::one());
+		let mut to_delete_block_number = first_saved_receipt;
+		while to_delete_block_number < new_first_saved_receipt {
+			let delete_block_number_key =
+				(EXECUTION_RECEIPT_BLOCK_NUMBER, to_delete_block_number).encode();
+			if let Some(hashes_to_delete) =
+				load_decode::<_, Vec<Block::Hash>>(backend, delete_block_number_key.as_slice())?
+			{
+				keys_to_delete.extend(
+					hashes_to_delete.into_iter().map(|h| (EXECUTION_RECEIPT_KEY, h).encode()),
+				);
+				keys_to_delete.push(delete_block_number_key);
+			}
+			to_delete_block_number = to_delete_block_number.saturating_add(One::one());
 		}
 
 		backend.insert_aux(
```

Comment:
This insert has the first argument repeated in both branches, maybe makes sense to move it out of the `if`, especially since code is a bit verbose.